server/raft_log.go

```diff
@@ -63,6 +80,35 @@ func newRaftLog(log logger.Logger, fileName string, sync bool, trailingLogs int)
 		simpleDelThresholdHigh: 100000,
 		codec:                  &codec.MsgpackHandle{},
 	}
+	if encrypt {
+		// Always check env variable first.
+		key := []byte(os.Getenv(stores.CryptoStoreEnvKeyName))
+		if len(key) == 0 {
+			key = encryptionKey
+			if len(key) == 0 {
+				return nil, stores.ErrCryptoStoreRequiresKey
+			}
+		}
+		defer func() {
+			for i := 0; i < len(encryptionKey); i++ {
+				encryptionKey[i] = 'x'
+			}
+		}()
+
+		h := sha256.New()
+		h.Write(key)
+		keyHash := h.Sum(nil)
+		gcm, err := chacha20poly1305.New(keyHash)
```

Comment:
Should this be settable longer term?

Reply:
Settable how? chacha20poly1305.New() requires a 256 bit key, which is what the few lines above provide.

Reply:
Would default to AES, faster on Intels and most server boxes. ChaCha good for ARM devices.

Reply:
Add one byte indicating algorithm to use..